pkg/management/component.go

```diff
@@ -391,7 +391,17 @@ func (c *component) genericCreateUser(ctx context.Context, accessToken string, c
 	user api.UserRepresentation, generateUsername bool, generateNameID bool, termsOfUse bool, useOnboardingCheckForExistingUser bool) (string, error) {
 	var userRep = api.ConvertToKCUser(user)
 	userRep.SetAttributeString(constants.AttrbSource, source)
-	userRep.SetAttributeString(constants.AttrbOnboardingStatus, managementOnboardingStatus)
+
+	realm := ctx.Value(cs.CtContextRealm).(string)
```

Comment:
I think we should use the targetRealmName. Realm from context could be "master" when you create a user into a customer realm